File: mail_profile.py
from datetime import datetime
from dataclasses import dataclass


@dataclass(frozen=True)
class MailProfile:
    message_id: str
    date_str: str
    date_dt: datetime | None
    link: str
    subj: str
    from_addr: str
    to_addr: str

    @property
    def excel_link(self) -> str:
        return f'=HYPERLINK("{self.link}","メール")' if self.link else ""

File: MailGrep/MailGrep/Models/MailProfile.swift
import Foundation

struct MailProfile: Identifiable, Hashable {
    let id = UUID()
    let messageID: String
    let dateStr: String
    let dateValue: Date?
    let link: String
    let subject: String
    let fromAddr: String
    let toAddr: String
    let emlxPath: URL

    var mailAppURL: URL? {
        guard !messageID.isEmpty else { return nil }
        let cleanID = messageID
            .trimmingCharacters(in: CharacterSet(charactersIn: "<>"))
        guard let encoded = cleanID.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) else {
            return nil
        }
        return URL(string: "message:\(encoded)")
    }
}
